src/profile.c

```diff
@@ -365,88 +365,113 @@ static inline Int8 getTicks(void)
+                    ticksDone = ticks;
+                }
+                else {
+                    ticksDone = (ticks / profileState.minimumProfileTick) *
+                                profileState.minimumProfileTick;
+                }
+                ticks -= ticksDone;
+                outputFilenameIdIfRequired(nameid);
+                fprintf(profileState.Stream,
+                        "{\"Type\":\"%c\",\"Ticks\":%d,\"Line\":%d,"
+                        "\"FileId\":%d}\n",
+                        exec ? 'E' : 'R', ticksDone, (int)line, (int)nameid);
+                profileState.lastOutputtedTime = newticks;
+                profileState.lastNotOutputted.line = -1;
+                profileState.lastOutputted.line = line;
+                profileState.lastOutputted.fileID = nameid;
+                profileState.lastOutputtedExec = exec;
+            }
+            else {
+                profileState.lastNotOutputted.line = line;
+                profileState.lastNotOutputted.fileID = nameid;
+            }
+        }
+        else {
+            outputFilenameIdIfRequired(nameid);
+            fprintf(profileState.Stream,
+                    "{\"Type\":\"%c\",\"Line\":%d,\"FileId\":%d}\n",
+                    exec ? 'E' : 'R', (int)line, (int)nameid);
+            profileState.lastOutputted.line = line;
+            profileState.lastOutputted.fileID = nameid;
+            profileState.lastOutputtedExec = exec;
+            profileState.lastNotOutputted.line = -1;
+        }
+    }
+}
+
 // exec : are we executing this statement
 // visit: Was this statement previously visited (that is, executed)
 static inline void outputStat(Stat stat, int exec, int visited)
 {
-  UInt line;
-  int nameid;
+    UInt line;
+    int  nameid;
 
-  CheckLeaveFunctionsAfterLongjmp();
+    // Explicitly skip these two cases, as they are often specially handled
+    // and also aren't really interesting statements (something else will
+    // be executed whenever they are).
+    if (TNUM_STAT(stat) == T_TRUE_EXPR || TNUM_STAT(stat) == T_FALSE_EXPR) {
+        return;
+    }
 
-  Int8 ticks = 0, newticks = 0;
+    CheckLeaveFunctionsAfterLongjmp();
 
-  // Explicitly skip these two cases, as they are often specially handled
-  // and also aren't really interesting statements (something else will
-  // be executed whenever they are).
-  if (TNUM_STAT(stat) == T_TRUE_EXPR || TNUM_STAT(stat) == T_FALSE_EXPR) {
-    return;
-  }
+    // Catch the case we arrive here and profiling is already disabled
+    if (!profileState_Active) {
+        return;
+    }
 
-  // Catch the case we arrive here and profiling is already disabled
-  if (!profileState_Active) {
-    return;
-  }
+    nameid = getFilenameIdOfCurrentFunction();
+    outputFilenameIdIfRequired(nameid);
 
-  nameid = getFilenameIdOfCurrentFunction();
-  outputFilenameIdIfRequired(nameid);
+    // Statement not attached to a file
+    if (nameid == 0) {
+        return;
+    }
```

Comment:
Wouldn't it make more sense to check `nameid==0` before calling `outputFilenameIdIfRequired`?

Ah, OK, I see it was like that before (and `outputFilenameIdIfRequired` does nothing if nameid is zero). Ah well,